esphome/components/rtttl/rtttl.h

```diff
@@ -31,6 +31,13 @@ class Rtttl : public Component {
 #ifdef USE_SPEAKER
   void set_speaker(speaker::Speaker *speaker) { this->speaker_ = speaker; }
 #endif
+  void set_gain(double gain) {
+    if (gain < 0.1)
+      gain = 0.1;
+    if (gain > 1)
+      gain = 1;
+    this->gain_ = gain;
+  }
```

Comment:
This could also use `clamp` from `helpers.h`
```suggestion
  void set_gain(float gain) {
    if (gain < 0.1f)
      gain = 0.1f;
    if (gain > 1.0f)
      gain = 1.0f;
    this->gain_ = gain;
  }
```